### DISCUSSION TITLE
/ddos-azure-functions-front-door/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#606] [Comments]
Someone is DDoS-ing my Azure Functions, and here is what you can do
        
[https://www.eliostruyf.com/ddos-azure-functions-front-door/](https://www.eliostruyf.com/ddos-azure-functions-front-door/)

**Comment by comment-migrator** [comment]
```header
Originally posted by TW McSheister on 2022-10-31 15:14
```

Great information and thanks for the detail, but doesn't the cost of Azure FD and WAF exceed your credits?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2022-11-01 10:36
```

Hi @twmcsheister:disqus, fair point you make, but I picked the Front Door standard offering, which is 35 EUR per month, and the associated WAF is included. At my credit renewal, I'll make the balance available.

**Comment by rvdwegen** [comment]
Are you still happy with the Azure Front Door solution? I'm looking at it for a similair rate limiting purpose.

**Reply by estruyf** [reply]
Hi @rvdwegen, I've had a couple of chats with AWS support to stop the bots overloading my service, but as the IPs change to rapid, it is hard for them it seems to find the source, and for each IP I should open a new support ticket. So I finally gave up, and I went for a cheaper option: Cloudflare.

My current flow is as follows:

```mermaid
flowchart LR
    User --> Cloudflare --> AFD(Azure Front Door) --> AF(Azure Function)
```

The nice thing about Cloudflare is that they have **Unmetered DDoS Protection**. With 2M calls daily to the Azure Front Door, the price kept increasing because I had to pay for the traffic. 

I configured a rate limit and some rules to block specific clients on Cloudflare. Azure Front Door is still in between the Azure Function, just to be sure, but also to not expose the Azure Function to the outside world.

**Reply by rvdwegen** [reply]
I haven't looked at Cloudflare in a while to be honest, I vaguely remember you have to have your DNS with them to make use of other services, is that (still?) the case?

**Reply by estruyf** [reply]
That is correct, you need to have your DNS on it as well as it is coupled to it. For me that made sense, as pricing wise, it was a lot cheaper.
